akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtoAnySerialization.scala

```diff
@@ -104,16 +111,36 @@ import scalapb.options.Scalapb
   import ProtoAnySerialization._
 
   private val serialization = SerializationExtension(system.classicSystem)
+  private lazy val akkaProtobufSerializer: Serializer =
+    serialization.serializerFor(classOf[com.google.protobuf.GeneratedMessageV3])
```

Comment:
yes, see https://github.com/akka/akka-projection/pull/959/files#r1302516904

btw, the `proto` binding is not only for "internal" use, it's for public use, so your binding in the sample is correct

This lookup is really just because I don't want to hardcode the serializer id. The optimizations will only take place for proto messages if they are bound to the default Akka ProtobufSerializer. We can't know what a custom protobuf serializer would do.

Reply:
My surprise was more that the proto3 serializer would understand how to serialise messages that are not `com.google.protobuf.GeneratedMessageV3` subtypes but I didn't dig deeper yet.

Reply:
Unrelated to this check though, so this is fine 👍

Reply:
yeah, it's using some naming conventions and reflection